pkg/elb/cleanup.go

```diff
@@ -0,0 +1,121 @@
+package elb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	awsprovider "k8s.io/kubernetes/pkg/cloudprovider/providers/aws"
+
+	"github.com/weaveworks/eksctl/pkg/eks"
+)
+
+const (
+	elbClassic = iota
+	elbV2      = iota
+)
+
+// CleanupLoadBalancers finds and deletes any dangling ELBs attached to a service
+func CleanupLoadBalancers(elbapi elbiface.ELBAPI, elbv2api elbv2iface.ELBV2API, client *eks.Client) error {
+	kubernetesCS, err := client.NewClientSet()
+	if err != nil {
+		return err
+	}
+	services, err := kubernetesCS.CoreV1().Services("").List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Delete Services of type 'LoadBalancer'
+	elbs := map[string]int{}
```

Comment:
I'd use [`k8s.io/apimachinery/pkg/util/sets`](https://godoc.org/k8s.io/apimachinery/pkg/util/sets), we use it in other places already for this purpose. I can see that you store them by type, but seems like using separate sets would be just a little more readable. Or a `map[int]sets.String`... Up to you, just a suggestion.

Reply:
I would rather not depend on kubernetes for data structures we can obtain directly from the standard library (even at a readability cost)